### PR TITLE
feat(WebXR): Add support for head-mounted augmented reality displays

### PR DESCRIPTION
**Pull request by tbirdso**
### Context

Adds support for rendering to WebXR head-mounted augmented reality sessions.

### Results

Previous behavior: vtk.js supported HMD VR (two cameras) or mobile AR (one camera) sessions, but not AR HMDs (two cameras). Side effect was that rendering in an AR headset (Meta Quest 2) would produce undefined behavior with only one eye able to see the scene at a time.

Updated behavior: Proper stereoscopic rendering for AR HMDs, combines existing support for VR HMD rendering with AR transparency.

Note that HoloLens 2 AR HMDs appear to be treated as VR headsets by WebXR and support `immersive-vr` but not `immersive-ar` sessions. Scale factors for VR scenes are updated to be more comfortable for HoloLens users.

Screenshot captured on HoloLens 2:

hololens-cropped

### Changes

- [x] Documentation and TypeScript definitions were updated to match those changes
- [x] Adds `HmdAR` enum value to request head-mounted augmented reality XR session
- [x] Adds handling for `HmdAR` session in `OpenGL/RenderWindow`
- [x] Moves scene background handling for AR sessions from application level into `OpenGL/RenderWindow` to reduce redundancy
- [x] Adds scripts to securely serve WebXR and WebGPU examples locally via HTTPS with self-signed SSL certificate

### PR and Code Checklist

- [x] [semantic-release](https://github.com/semantic-release/semantic-release) commit messages
- [x] Run `npm run reformat` to have correctly formatted code

### Testing

- [x] Tested environment:
  - **vtk.js**:  `master`
  - **OS**:  Windows 10
  - **Browser**:  Microsoft Edge

Tested on the following devices:
- Meta Quest 2 (w/ AR passthrough)
- Microsoft HoloLens 2
- Samsung Galaxy (mobile AR)

### CODE REVIEW

**Comment by tbirdso** [comment]
cc @LucasGandel

File: Examples/Volume/WebXRVolume/index.md [hunk 1]
```diff
@@ -14,12 +14,19 @@ Volume rendering can be processor-intensive. Smaller volumes may give better per
 
 ### Looking Glass Holographic Support
 
-Holographic scenes can be rendered to a Looking Glass display by specifying the `xrSessionType` in the example URL.
+Holographic scenes can be rendered to a Looking Glass display by specifying `xrSessionType=2` in the example URL.
```
[the remark this answers is not in the document]
Comment (tbirdso) [reply Examples/Volume/WebXRVolume/index.md:17]: I'm in favor of the proposed change. Given that the `xrSessionType` URL parameter is available at the application level in several examples, introduced in https://github.com/Kitware/vtk-js/commit/2ce9479ccf4d00ade3fa01d5bda7241e2ca40b50 and in https://github.com/Kitware/vtk-js/commit/92057802ad0bfce28022ad4160dd290f0272108e, I would rather introduce that change in a separate PR and keep this PR focused on AR HMD support.

File: Examples/Geometry/AR/index.js [hunk 2]
```diff
@@ -19,6 +20,15 @@ import '@kitware/vtk.js/IO/Core/DataAccessHelper/JSZipDataAccessHelper';
 
 import controlPanel from './controller.html';
 
+// ----------------------------------------------------------------------------
+// Parse URL parameters
+// ----------------------------------------------------------------------------
+const userParams = vtkURLExtract.extractURLParameters();
+const requestedXrSessionType =
+  userParams.xrSessionType !== undefined
+    ? userParams.xrSessionType
```
[the remark this answers is not in the document]
Comment (tbirdso) [reply Examples/Geometry/AR/index.js]: Updated

**Comment by djelouze** [comment]
Thank you so much!!

**Comment by floryst** [comment]
LGTM!

**Comment by djelouze** [comment]
Thank you so much!

**Pull request merged**